### DISCUSSION TITLE
limit inventory

### DISCUSSION THREAD

**Discussion by hmoffatt** [semaphoreui/semaphore#1089] [Q&A] [answered]
I have all my hosts in a single hosts.ini file in my repository and I can use it in a task template. But I'd like to be able to easily run a task on a sub-set of hosts, using ansible-playbook's --limit option.

Is there a way to use this option from Semaphore?

Ideally you could create easy add a limit in a task template, or maybe be able to add a limit into an existing inventory.

**Comment by hmoffatt** [comment]
It would also be helpful to allow a task to run against multiple inventory sources. ansible-playbook allows this with multiple `-i` options.

**Comment by ansibleguy** [comment, marked as answer]
Yes - you can supply the limit inside the task or task-template JSON field:

```json
["--limit", "host1,host2"]
```

Also: I opened a specific feature-request for adding a `--limit` field: https://github.com/ansible-semaphore/semaphore/issues/1357

Regarding multiple inventories - you could try to also supply them using the 'JSON' field as those variables seem to be directly passed to Ansible.
